### PR TITLE
[WIP] libuv

### PR DESCRIPTION
**Pull request by majestrate**

### CODE REVIEW

File: llarp/ev/ev_libuv.hpp [hunk 1]
```diff
@@ -0,0 +1,104 @@
...
+{
+  struct Loop : public llarp_ev_loop
+  {
+    bool
+    init() override;
+
+    int
+    run() override
+    {
+      return -11;
```
Comment (michael-loki) [inline llarp/ev/ev_libuv.hpp]: why -11 here?

File: llarp/ev/ev_libuv.hpp [hunk 2]
```diff
@@ -0,0 +1,104 @@
+#ifndef LLARP_EV_LIBUV_HPP
+#include <ev/ev.hpp>
+#include <uv.h>
+#include <vector>
+#include <functional>
+
+namespace libuv
+{
+  struct Loop : public llarp_ev_loop
```
Comment (michael-loki) [inline llarp/ev/ev_libuv.hpp]: flag as final here?

File: llarp/ev/ev_libuv.hpp (hunk 1 above)
Comment (majestrate) [reply llarp/ev/ev_libuv.hpp]: typo

File: llarp/CMakeLists.txt [hunk 3]
```diff
@@ -75,6 +75,7 @@ set(LIB_PLATFORM_SRC
 # tun
   ${LIBTUNTAP_SRC}
   ${EV_SRC}
+  ev/ev_libuv.cpp
```
Comment (despair86) [inline llarp/CMakeLists.txt]: should also be placed in a not-windows block

File: llarp/utp/linklayer.cpp [hunk 4]
```diff
@@ -43,73 +43,7 @@ namespace llarp
...
-        int err = WSAGetLastError();
-        FormatMessage(FORMAT_MESSAGE_FROM_SYSTEM, nullptr, err, LANG_NEUTRAL,
-                      buf, 1024, nullptr);
-        LogError("sendto failed: ", buf);
-#else
-        LogError("sendto failed: ", strerror(errno));
-#endif
-      }
-      return 0;
+      return l->m_udp.sendto(&l->m_udp, arg->address, arg->buf, arg->len);
```
Comment (despair86) [inline llarp/utp/linklayer.cpp]: duly noted, will patch that into the win32 event loops

File: llarp/ev/ev.h [hunk 5]
```diff
@@ -27,6 +27,8 @@ typedef SSIZE_T ssize_t;
 #include <stdint.h>
 #include <stdlib.h>
 
+#include <uv.h>
```
Comment (despair86) [inline llarp/ev/ev.h]: oops

Comment (majestrate) [reply llarp/ev/ev.h]: wrap it in a `#if !defined(WIN32)` ?

Comment (despair86) [reply llarp/ev/ev.h]: ye

**Pull request merged**